src/flags.zig

```diff
@@ -302,7 +302,18 @@ fn parse_value(comptime T: type, flag: []const u8, value: [:0]const u8) T {
     comptime unreachable;
 }
 
-pub const ByteSize = struct { bytes: u64 };
+pub const ByteSize = struct {
+    bytes: u64,
+    original: []const u8,
+
+    pub fn default(value: u64) ByteSize {
+        return .{
+            .bytes = value,
+            .original = "error", // this shouldn't ever be visible in the UI
+        };
+    }
+};
+
```

Comment:
I think we can be more precise here, making this type responsible for parsing the user input, so we don't need to store the original string or handle special cases for constants.

For example:

```zig
pub const ByteUnit = enum(u32) {
    bytes = 1,
    kb = 1024,
    mb = .kb * 1024,
    gb = .mb * 1024,
    tb = .gb * 1024,
};

pub const ByteSize = struct {
    value: u64,
    unit: ByteUnit,

    pub fn parse(input: []const u8) ByteSize {
        // Move the "parse_value_size" logic here.
        return .{
            .value = parsed_value,
            .unit = parsed_unit,
        };
    }

    pub fn bytes(size: *const ByteSize) u64 {
        return size.value * @intFromEnum(size.unit);
    }

    pub fn suffix(size: *const ByteSize) []const u8 {
        return switch (size.unit) {
            .bytes => "",
            .kb => "KB",
            .mb => "MB",
            .gb => "GB",
            .tb => "TB",
        };
    }
};

// Printing an error message:
flags.fatal("--limit-storage: size {}{s} is below minimum: {}MB", .{
                    start.limit_storage.value,
                    start.limit_storage.suffix(),
                    @divExact(storage_size_limit_min, 1024 * 1024),
                });
```

Reply:
I have made this change and pushed it. Some of my changes may not be idiomatic so please give it another look.

e.g. I wasn't able to write a self-referential enum like the one you suggested. I also used an underscore to disambiguate a name.

Reply:
My bad! TIL that [zig does not allow self-dependency on enums](https://github.com/ziglang/zig/issues/5850), which was surprising since it's not an autonumerated enum.